wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,491 @@
+    const { keplr } = window;
+    const offlineSigner = await keplr.getOfflineSignerOnlyAmino(
+      ibcAsset.chainInfo.chainId,
+    );
+    const accounts = await offlineSigner.getAccounts();
+    setRemoteChainAddress(accounts[0].address);
+    setRemoteChainSigner(offlineSigner);
+  };
+
+  const isAmountInvalid = useMemo(() => {
+    switch (direction) {
+      case IbcDirection.Withdrawal:
+        try {
+          const val = parseAsValue(
+            amount,
+            purse?.displayInfo?.assetKind,
+            purse?.displayInfo?.decimalPlaces,
+          );
+          return val === 0n || val > (purseBalance ?? 0n);
+        } catch {
+          return true;
+        }
+      case IbcDirection.Deposit:
+        try {
+          const val = parseAsValue(
+            amount,
+            purse?.displayInfo?.assetKind,
+            purse?.displayInfo?.decimalPlaces,
+          );
+          return val === 0n || val > (remoteChainBalance ?? 0n);
+        } catch {
+          return true;
+        }
+    }
+  }, [amount, purse, purseBalance]);
+
+  const close = () => {
+    setAmount('');
+    setRemoteChainAddress('');
+    setRemoteChainSigner(null);
+    setError('');
+    handleClose();
+  };
+
+  const send = async () => {
+    setError('');
+
+    let val;
+    try {
+      val = String(
+        parseAsValue(
+          amount,
+          purse?.displayInfo?.assetKind,
+          purse?.displayInfo?.decimalPlaces,
+        ),
+      );
+    } catch (e) {
+      setError(String(e));
+    }
+
+    assert(ibcAsset);
+
+    if (direction === IbcDirection.Deposit) {
+      assert(remoteChainSigner && keplrConnection);
+      const { chainInfo, deposit } = ibcAsset;
+      const { sourceChannel, sourcePort, denom, gas } = deposit;
+      assert(gas);
+
+      setInProgress(true);
+      const signer = await SigningStargateClient.connectWithSigner(
+        chainInfo.rpc,
+        remoteChainSigner,
+      );
+      try {
+        const res = await signer.sendIbcTokens(
+          remoteChainAddress,
+          keplrConnection.address,
+          {
+            amount: val,
+            denom: denom,
+          },
+          sourcePort,
+          sourceChannel,
+          undefined,
+          timeoutTimestampSeconds(),
+          {
+            amount: [{ amount: '0', denom }],
+            gas,
+          },
+        );
+        close();
+        showSnackbar(
+          <>
+            Successfully executed transaction{' '}
+            <Link
+              color="rgb(0, 176, 255)"
+              href={`https://bigdipper.live/${chainInfo.explorerPath}/transactions/${res.transactionHash}`}
+              target={res.transactionHash}
+            >
+              ...{res.transactionHash.slice(res.transactionHash.length - 12)}
+            </Link>
+          </>,
+        );
+      } catch (e) {
+        setError(String(e));
+      } finally {
+        setInProgress(false);
+      }
+    } else {
+      /* Is Withdrawal */
```

Comment:
comments like this suggest breaking out a named function

Reply:
Done, but kept the comment